File: test/selenium/selenium-js-mocha/test/baseTest.js
import { Builder } from 'selenium-webdriver';

export async function createDriver() {
  const url = process.env.REMOTE_WEBDRIVER_URL;
  
  if (!url) throw new Error('REMOTE_WEBDRIVER_URL not set');
  const browser = process.env.BROWSER || 'chrome';
  
  return new Builder().usingServer(url).forBrowser(browser).build();
}

File: test/selenium/selenium-java/src/test/java/BaseTest.java
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.remote.RemoteWebDriver;
import org.openqa.selenium.remote.DesiredCapabilities;

import java.net.URL;

public abstract class BaseTest {

    protected WebDriver driver;

    @BeforeEach
    public void setUp() throws Exception {
        String remoteUrl = System.getenv("REMOTE_WEBDRIVER_URL");
        if (remoteUrl == null || remoteUrl.isBlank()) {
            throw new IllegalArgumentException("REMOTE_WEBDRIVER_URL ENV not set");
        }

        String browser = System.getenv().getOrDefault("BROWSER", "chrome");
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setBrowserName(browser);
        driver = new RemoteWebDriver(new URL(remoteUrl), capabilities);
    }

    @AfterEach
    public void tearDown() {
        if (driver != null) {
            driver.quit();
        }
    }
}